### PR TITLE
Add unknown scenario

### PR DESCRIPTION
**Pull request by gopipatell**
- Implemented unknown scenario
- Added test cases
- Moved TxFlow to service layer
- Added favicon

### CODE REVIEW

File: src/services/spiService/index.ts [hunk 1]
```diff
@@ -325,10 +325,53 @@ class SpiService {
...
+          type: detail.Type,
+          displayMessage: detail.DisplayMessage,
+          amountCents: 0,
+          awaitingSignatureCheck: detail.AwaitingSignatureCheck,
+          finished: detail.Finished,
+          success: detail.Success,
+          response: {
+            data: {
+              rrn: detail?.Response?.Data.rrn,
+              schemeAppName: detail?.Response?.Data.scheme_app_name as string,
```
Comment (mx51damon) [inline src/services/spiService/index.ts:340]: I was thinking we can remove `as string` for this line

File: src/services/spiService/index.ts [hunk 2]
```diff
@@ -325,10 +325,53 @@ class SpiService {
...
+          id: detail.Id,
+          type: detail.Type,
+          displayMessage: detail.DisplayMessage,
+          amountCents: 0,
+          awaitingSignatureCheck: detail.AwaitingSignatureCheck,
+          finished: detail.Finished,
+          success: detail.Success,
+          response: {
+            data: {
+              rrn: detail?.Response?.Data.rrn,
```
Comment (mx51damon) [inline src/services/spiService/index.ts:339]: I was thinking do we need to add optional chaining operator for all `Data` object? (I am sorry, previously I didn't cover this part.)

**Review by mx51damon — CHANGES_REQUESTED** [review]
Hi @gopipatell , I have put quite a few comments (I hope you don't mind) and welcome to have a discussion so we can have a chat and share some thoughts about the code logics in detail.

File: src/components/UnknownTransactionModal/index.styles.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      padding: '1rem',
+      color: theme.palette.text.primary,
+    },
+    modalSublHeading: {
+      fontWeight: 500,
+      fontSize: '1.125rem',
+      color: theme.palette.text.secondary,
+    },
+    modalBtn: {
+      margin: theme.spacing(3, 1, 1, 1),
```
Comment (mx51damon) [inline src/components/UnknownTransactionModal/index.styles.ts:17]: might suggest the shortcut as `theme.spacing(3, 1, 1),`

File: src/components/OrderConfirmation/index.tsx [hunk 4]
```diff
@@ -51,20 +54,26 @@ function OrderConfirmation({ title, pathname, currentAmount }: IOrderConfirmatio
...
-  const subtotalAmount: number = useSelector(productSubTotalSelector);
   const tipAmount: number = useSelector(orderTipAmountSelector);
   const cashoutAmount: number = useSelector(orderCashoutAmountSelector);
   const terminals = useSelector(pairedConnectedTerminalList);
+  const [subtotalAmount, setSubtotalAmount] = useState<number>(useSelector(productSubTotalSelector));
   const selectedTerminal = useSelector(selectedTerminalIdSelector);
   const currentTerminal = useSelector(terminalInstance(selectedTerminal)) as ITerminalProps;
 
+  const isFinished = currentTerminal?.txFlow?.finished ?? false;
+  const successStatus = currentTerminal?.txFlow?.success;
```
[the remark this answers is not in the document]
Comment (gopipatell) [reply src/components/OrderConfirmation/index.tsx:66]: can't use `terminalTxFlowSuccessTracker` as I need unknown status

[the remark this answers is not in the document]
Comment (mx51damon) [reply src/components/OrderConfirmation/index.tsx:66]: 👌

**Review by mx51damon — APPROVED** [review]
Thank you Gopi, nice work ~~

**Review by mx51damon — APPROVED** [review]

**Pull request merged**